### PR TITLE
update Rails to 5.1 initial code drop

### PR DESCRIPTION
**Pull request by akostadinov**
**What this PR does / why we need it**:

This branch has Rails 5.1 updated at a state where test suite runs(but many tests fail). We can use it as a base to collaborate on fixing all these test cases.

The basic idea is that we can pick failing test cases to fix, update deprecated code and gems.

Ideally all such updates will be backported to master and merged. This branch should be rebased on master regularly.

Because most changes should already be backported, at the end this PR should not be long and all test cases should pass. Then we can merge and consider the task completed.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-5010

**Verification steps** 

Test suite passes.

### CODE REVIEW

File: app/controllers/admin/api/cms/templates_controller.rb [hunk 1]
```diff
@@ -55,7 +56,7 @@ def index
   ##=  }
   ##=
   def create
-    type = cms_template_params.delete('type')
+    type = params.require('type')
```
Comment (jsmadis) [inline app/controllers/admin/api/cms/templates_controller.rb:59]: @akostadinov Do you remember why did you change it like this? It doesn't make sense to me

Comment (akostadinov) [reply app/controllers/admin/api/cms/templates_controller.rb:59]: probably some kind of mistake on my part and no second thought about it

I can fix it, or do you work on a PR that already fixes it?

Comment (jsmadis) [reply app/controllers/admin/api/cms/templates_controller.rb]: No need, already pushed the fix for it

File: app/controllers/provider/admin/dashboard/potential_upgrades_controller.rb [hunk 2]
```diff
@@ -24,8 +24,8 @@ def current_upgrades
   def is_set_up_correctly?
     usage_limits = current_account
       .application_plans.joins(:usage_limits)
-      .grouping{ issuer_id }.unscope(:order)
-      .references(:usage_limits).count{ usage_limits.id }
+      .grouping { issuer_id }.unscope(:order)
+      .references(:usage_limits).select { id }.count
```
Comment (jsmadis) [inline app/controllers/provider/admin/dashboard/potential_upgrades_controller.rb:28]: This one is weird, the error is: 
```
Mysql2::Error: Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column '3scale_system_test.plans.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by: SELECT `plans`.* FROM `plans` INNER JOIN `usage_limits` ON `usage_limits`.`plan_id` = `plans`.`id` AND `usage_limits`.`plan_type` = 'Plan' INNER JOIN `services` ON `plans`.`issuer_id` = `services`.`id` WHERE `plans`.`type` IN ('ApplicationPlan') AND `services`.`account_id` = 258 AND `plans`.`issuer_type` = 'Service' GROUP BY `plans`.`issuer_id`
```

Rails 5.0 generates this query:

```sql
SELECT COUNT(*) AS count_all, `plans`.`issuer_id` AS plans_issuer_id FROM `plans` INNER JOIN `usage_limits` ON `usage_limits`.`plan_id` = `plans`.`id` AND `usage_limits`.`plan_type` = 'Plan' INNER JOIN `services` ON `plans`.`issuer_id` = `services`.`id` WHERE `plans`.`type` IN ('ApplicationPlan') AND `services`.`account_id` = 429 AND `plans`.`issuer_type` = 'Service' AND `plans`.`type` IN ('ApplicationPlan') AND `plans`.`type` IN ('ApplicationPlan') GROUP BY `plans`.`issuer_id`
```

New query generates this:
```sql
SELECT COUNT(`plans`.`id`) AS count_all, `plans`.`issuer_id` AS plans_issuer_id FROM `plans` INNER JOIN `usage_limits` ON `usage_limits`.`plan_id` = `plans`.`id` AND `usage_limits`.`plan_type` = 'Plan' INNER JOIN `services` ON `plans`.`issuer_id` = `services`.`id` WHERE `plans`.`type` IN ('ApplicationPlan') AND `services`.`account_id` = 435 AND `plans`.`issuer_type` = 'Service' GROUP BY `plans`.`issuer_id`
```

They are the same, with the exception of condition ``plans`.`type` IN ('ApplicationPlan')` which is only used once as it should be

**Comment by jsmadis** [comment]
@akostadinov sure, but you have a typo there

```
statement_limit: <%= ENV.fetch('ORACLE_STATEMENT_LIMIT', 250) %>
```
